packages/core/src/baseCogniteClient.ts

```diff
@@ -35,7 +35,7 @@ export interface ClientOptions {
   /** URL to Cognite cluster, e.g 'https://greenfield.cognitedata.com' **/
   baseUrl?: string;
   project: string;
-  getToken: () => Promise<string>;
```

Comment:
How does this work with the "OIDC authentication using code authorization w pkce" example? Afaik for token refresh to work the getToken needs to be able to resolve a new token if the old one has expired. Without promises here I don't think it is possible to write a single function that can resolve a new access token externally as the function must resolve immediately. 

Case where this is important: Long-lived clients authenticated via client credentials such as APIs

Reply:
You're right, but, the problem are exactly here when you try to get logged using api-keys using only promises without async/await. I did solve this problem on the new feature that will be on the next release, using the new auth-wraper that will do automatically the token refresh and of course the api key on this way too. Maybe we can discuss more about in a meeting, please text-me on slack and we can schedule.